qiskit/transpiler/passes/routing/algorithms/bip_model.py

```diff
@@ -397,15 +410,24 @@ def _max_expected_fidelity(self, node, i, j):
             for k, gfid in enumerate(self._gate_fidelities(node))
         )
 
-    def _max_expected_mirroed_fidelity(self, node, i, j):
+    def _max_expected_mirrored_fidelity(self, node, i, j):
         return max(
             gfid * self._cx_fidelity(i, j) ** k
             for k, gfid in enumerate(self._mirrored_gate_fidelities(node))
         )
 
     def _cx_fidelity(self, i, j) -> float:
         # fidelity of cx on global physical qubits
-        return 1.0 - self.bprop.gate_error("cx", [self.global_qubit[i], self.global_qubit[j]])
+        if self.bprop is not None:
+            try:
+                error_rate = self.bprop.gate_error(
+                    "cx", [self.global_qubit[i], self.global_qubit[j]]
+                )
+            except (AttributeError, KeyError, ValueError):
+                error_rate = self.default_cx_error_rate
+            return 1.0 - error_rate
+        else:
+            return 1.0 - self.default_cx_error_rate
```

Comment:
Any reason we have to catch those runtime errors here? I think branching based on whether `self.bprop is None` or not would be better, i.e. raising runtime errors when some problem happens in reading backend property.

Reply:
Ok, I reverted to using bprop if not None, without catching exceptions.